bin/yaml/libraries.yaml

```diff
@@ -2143,14 +2143,15 @@ libraries:
         - ./configure
       type: github
     proxy:
-      check_file: proxy.h
-      method: clone_branch
+      build_type: cmake
```

Comment:
will the previous targets (3.0.0 through 3.2.1) work with this build_type? If so: great, but we'll need to force build and install them too. If not, can you only change the new one (I can show how if that's the case)

Reply:
Yes, all the versions work with CMake. Although the include path will change (from "proxy.h" to "proxy/proxy.h"), I don't feel it's a big issue.

Reply:
Oh, that’s a valid concern. In this case, it’s probably better only enable CMake in nightly builds and future versions. Could you please show me any references for this scenario?

Reply:
I tried to override the build_type for older versions. Could you help review again?